test/clj/game/cards/upgrades_test.clj

```diff
@@ -3336,3 +3336,243 @@
        (click-card state :corp (refresh cor))
        (is (empty? (get-program state)) "Corroder uninstalled")
        (is (= "Corroder" (:title (last (:deck (get-runner))))) "GoCorroderrdian on bottom of Stack")))))
+
+(deftest oaktown-grid
+  (do-game
+    (new-game {:corp {:hand ["Oaktown Grid" "PAD Campaign"]}
+               :runner {:credits 7}})
+    (play-from-hand state :corp "Oaktown Grid" "New remote")
+    (play-from-hand state :corp "PAD Campaign" "Remote 1")
+    (let [og (get-content state :remote1 0)
+          pad (get-content state :remote1 1)]
+      (rez state :corp og)
+      (rez state :corp pad)
+    (take-credits state :corp)
+    (run-on state "Server 1")
+    (run-continue state)
+    (click-card state :runner pad)
+    (click-prompt state :runner "Pay 7 [Credits] to trash")
+    (is (= "PAD Campaign" (:title (first (:discard (get-corp))))) "PAD Campaign trashed for 7c (3 more than normal cost thanks to Oaktown Grid")
+    (is (= 0 (:credit (get-runner)))))))
+
+(deftest rutherford-grid
+  (do-game
+    (new-game {:corp {:hand ["Rutherford Grid" "Caduceus"]}})
+    (play-from-hand state :corp "Rutherford Grid" "New remote")
+    (play-from-hand state :corp "Caduceus" "Remote 1")
+    (let [rg (get-content state :remote1 0)
+          caduceus (get-ice state :remote1 0)]
+      (rez state :corp rg)
+      (rez state :corp caduceus)
+      (take-credits state :corp)
+      (run-on state "Server 1")
+      (run-continue state)
+      (fire-subs state caduceus)
+      (click-prompt state :corp "3")                        ;;Boost trace by 3
+      (is (= 8 (:strength (get-prompt state :runner))) "3 base, +2 of upgrade +3 boost"))))
```

Comment:
Probably best to not pump the trace, so the card's ability can be more easy seen.